data_measurements/dataset_statistics.py

```diff
@@ -542,6 +573,45 @@ def prepare_text_duplicates(self):
             )
             self.dup_counts_df[OUR_TEXT_FIELD] = self.dup_counts_df.index.copy()
 
+    def prepare_text_perplexities(self):
+        if not self.live:
+            perplexities = {PERPLEXITY_FIELD: [], OUR_TEXT_FIELD: []}
+            if self.gpt2_tokenized_df is None:
+                self.load_or_prepare_gpt2_tokenized_df()
+            rows = list(self.gpt2_tokenized_df.iterrows())
+            for i in tqdm(range(0, len(rows), _PERPLEXITY_BATCH_SIZE)):
+                row_batch = rows[i:i+_PERPLEXITY_BATCH_SIZE]
+                input_ids_batch = []
+
+                for index, row in row_batch:
+                    perplexities[OUR_TEXT_FIELD].append(row[OUR_TEXT_FIELD])
+                    # Add the begginging of sentence token to the input ids, so that we can get a
+                    # probability for the first word in the text from GPT2.
+                    input_ids_batch.append([_GPT2_TOKENIZER.bos_token_id] + row[TOKENIZED_FIELD])
```

Comment:
I'm drawing attention to this line, in case anyone thinks I should do something different. Another option could be to not include the bos token, but this would make perplexities of single word sentences undefined.